src/sinks/util/encoding/mod.rs

```diff
@@ -104,12 +110,24 @@ pub trait Encoder<T> {
     }
 }
 
-impl<E, T> Encoder<T> for Arc<E>
-where
-    E: Encoder<T>,
-{
-    fn encode_input(&self, input: T, writer: &mut dyn io::Write) -> io::Result<usize> {
-        (**self).encode_input(input, writer)
+#[cfg(feature = "codecs")]
+impl Encoder<Vec<Event>> for (Transformer, crate::codecs::Encoder<()>) {
+    fn encode_input(&self, events: Vec<Event>, writer: &mut dyn io::Write) -> io::Result<usize> {
```

Comment:
It's not super obvious to me how this is related to the primary change.

Reply:
Agree that this wasn't super obvious 😅 I factored that part out into a separate PR: https://github.com/vectordotdev/vector/pull/12178.

This has just been a piece that is used by many sinks that use `StandardEncodings`.